Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -112,6 +116,9 @@ class BrowserViewController: UIViewController, BrowserToolbarDelegate, TabManage
                 if let progress = change[NSKeyValueChangeNewKey] as Float? {
                     toolbar.updateProgressBar(progress)
                 }
+
+                toolbar.updateBackStatus(object.canGoBack)
+                toolbar.updateFowardStatus(object.canGoForward)
```

Comment:
Think I have it resolved. To clarify, in the previous comments we both referenced the `webViewDidStartLoading` method which is part of the UIWebViewDelegate protocol but we obviously both meant to reference the WKNavigationDelegate protocol method ;)

The method in that protocol I remembered trying was `webView:didStartProvisionalNavigation` and the problem was that the webView's `canGoBack` property does not seem to be updated at the time this delegate method is called (not sure if this is by design or not, but it is the observed behavior). However, it does seem up to date by the time `webView:didCommitNavigation` is called, so I will update the button statuses there and update the pull request.